src/sqlfluff/dialects/dialect_snowflake.py

```diff
@@ -1216,6 +1218,127 @@ class CopyOptionsSegment(BaseSegment):
     )
 
 
+@snowflake_dialect.segment(replace=True)
+class CreateSchemaStatementSegment(BaseSegment):
+    """A `CREATE SCHEMA` statement.
+
+    https://docs.snowflake.com/en/sql-reference/sql/create-schema.html
+    """
+
+    type = "create_schema_statement"
+    match_grammar = Sequence(
+        "CREATE",
+        Ref("OrReplaceGrammar", optional=True),
+        Ref("TemporaryTransientGrammar", optional=True),
+        "SCHEMA",
+        Ref("IfNotExistsGrammar", optional=True),
+        Ref("SchemaReferenceSegment"),
+        Sequence("WITH", "MANAGED", "ACCESS", optional=True),
+        AnyNumberOf(Ref("SchemaObjectParamsSegment")),
+        Sequence(
+            Sequence("WITH", optional=True),
+            "TAG",
+            Bracketed(
+                Delimited(
+                    Ref("NakedIdentifierSegment"),
+                    Ref("EqualsSegment"),
+                    Ref("QuotedIdentifierSegment"),
+                )
+            ),
+            optional=True,
+        ),
+    )
+
+
+@snowflake_dialect.segment()
+class AlterSchemaStatementSegment(BaseSegment):
+    """An `ALTER SCHEMA` statement.
+
+    https://docs.snowflake.com/en/sql-reference/sql/alter-schema.html
+
+    """
+
+    type = "alter_schema_statement"
+    match_grammar = Sequence(
+        "ALTER",
+        "SCHEMA",
+        Sequence("IF", "EXISTS", optional=True),
+        Ref("SchemaReferenceSegment"),
+        OneOf(
+            Sequence(
+                "RENAME",
+                "TO",
+                Ref("SchemaReferenceSegment"),
+            ),
+            Sequence(
+                "SWAP",
+                "WITH",
+                Ref("SchemaReferenceSegment"),
+            ),
+            Sequence(
+                "SET",
+                AnyNumberOf(
+                    Ref("SchemaObjectParamsSegment"),
+                ),
+            ),
+            Sequence(
+                "UNSET",
+                AnyNumberOf(
+                    Delimited(
+                        "DATA_RETENTION_TIME_IN_DAYS",
+                        "MAX_DATA_EXTENSION_TIME_IN_DAYS",
+                        "DEFAULT_DDL_COLLATION",
+                        "COMMENT",
+                    ),
+                    Sequence("TAG", Delimited(Ref("NakedIdentifierSegment"))),
+                    min_times=1,
+                ),
+            ),
+            Sequence(OneOf("ENABLE", "DISABLE"), Sequence("MANAGED", "ACCESS")),
+        ),
+    )
+
+
+@snowflake_dialect.segment()
+class SchemaObjectParamsSegment(BaseSegment):
+    """A Snowflake Schema Object Param segment.
+
+    https://docs.snowflake.com/en/sql-reference/sql/create-schema.html
+    https://docs.snowflake.com/en/sql-reference/sql/alter-schema.html
+    """
+
+    type = "schema_object_properties"
+
+    match_grammar = AnyNumberOf(
```

Comment:
This could be OneOf(), as it's wrapped in AnyNumberOf in the segments that reference it

Reply:
Good point, but I'd actually prefer to go the other way around: keep the `AnyNumberOf()` here and remove it in the referencing segments.

Reply:
Okay, happy for you to go with that - based on it being order-dependent in the referencing segment structure